### ISSUE TITLE
Lint/NestedMethodDefinition shouldn't be raised inside an eval block

### ISSUE THREAD

**Issue by segiddins** [bbatsov/rubocop#2456]
The following shouldn't cause any violations

```ruby
class C
  def x
    C.class_eval do
      def y
      end
    end
  end
end
```

**Comment by alexdowad** [comment]
Makes sense.
